### ISSUE TITLE
Start working with the matching script

### ISSUE THREAD

**Issue by jgbarah** [priya299/outreachy-project#7]
When you master Perceval backends for git and mailing lists, let's start to work with the matching of threads to commits. For that, let's use the code in [@dicortazar repo for the Xen analysis](https://github.com/dicortazar/xen-code-review-analysis). In particular, the matching code is in the [data-retrieval directory](https://github.com/dicortazar/xen-code-review-analysis/tree/master/data-retrieval).

Please, go through the README.md, undestand the process, and go through the script. The target is to do basically the same (find commits that match threads related to patches in the mailing list), but using the ElasticSearch databases produced with Perceval, instead than the ones produced with MetricsGrimoire, which is what Dani did.

When you understand the process, please define roughly the algorithm you intend to use for the matching, based on the data available from Perceval, the threading algorithm you implemented, and want you understand from Dani's procedure and script.

Please, use this ticket for any clarification you may need, and for describing the algorithm you envision. We can discuss and polish it here, before we start coding (we will open another ticket for that).

**Comment by ghost** [comment]
I think this is a good approach: we should definitely document the existing matching algorithm. I want to have a look at it and see how well that matches reality and whether it can be improved. If there is scope for improvements I suggest we create some additional tickets later.
